Sources/Core/ConditionalCompilationConfiguration.swift

```diff
@@ -0,0 +1,35 @@
+import Utils
+
+/// The configuration needed for conditional compilation.
+public struct ConditionalCompilationConfiguration: Codable {
```

Comment:
```suggestion
public struct ConditionalCompilationFactors: Codable {
```

I was racking my brains for this word today and eventually everything became clear.  The name `ConditionalCompilationConfiguration` suggests this thing configures conditional compilation.  I have to stretch the usual sense in which we talk about configuring a language feature quite far in order to make that work.  For a while I thought the word might be `Criteria`, but a criterion would be something like `hyloVersion > 1.0.1` and this does not contain that.  So I was looking for a word meaning “a piece of information based on which a decision is made.”  The definitions I can find for “Factor” fit that idea perfectly, c.f.
- https://dictionary.cambridge.org/us/dictionary/english/factor (1)
- https://www.merriam-webster.com/dictionary/factor (2a)
- https://www.collinsdictionary.com/us/dictionary/english/factor (1)
- https://www.dictionary.com/browse/factor (1)

Reply:
Fixed.